p2panda-rs/src/test_utils/mocks/node.rs

```diff
@@ -200,6 +200,57 @@ impl Node {
         self.db.clone()
     }
 
+    /// Add an entry to the db, has no effect if entry already exists.
+    fn insert_entry(&mut self, entry: &LogEntry, document_id: &Hash) -> bool {
+        let author = entry.author();
+
+        if let Some(author_logs) = self.db.get_mut(&author) {
+            if let Some(document_log) = author_logs.get_log_mut_by_document_id(document_id) {
+                let existing_entry = document_log
+                    .entries()
+                    .into_iter()
+                    .find(|ex_entry| ex_entry.hash() == entry.hash());
+
+                if existing_entry.is_none() {
+                    document_log.add_entry(entry.to_owned());
+                    return true;
+                }
+                false
+            } else {
+                author_logs.create_new_log(
+                    document_id.to_owned(),
+                    &entry.entry_encoded(),
+                    &entry.operation_encoded(),
+                );
+                true
+            }
+        } else {
+            let mut author_logs = AuthorLogs::new();
+            author_logs.create_new_log(
+                document_id.to_owned(),
+                &entry.entry_encoded(),
+                &entry.operation_encoded(),
+            );
+            self.db.insert(author.as_str().into(), author_logs);
+            true
+        }
+    }
+
+    /// Replicate entries from other_node to this node.
+    pub fn replicate_with(&mut self, other_node: &mut Node) -> Vec<Hash> {
```

Comment:
Could be nice to have examples for mock node methods. I know they are also in the tests so not required, just - nice to have :-)